pkg/agent/util/iptables/lock.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iptables
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"golang.org/x/sys/unix"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	xtablesLockFilePath       = "/var/run/xtables.lock"
+	xtablesLockFilePermission = 0600
+)
+
+type xtablesLock struct {
+	// For testing purpose.
+	lockFilePath string
+	lockFile     *os.File
+}
+
+func NewXTablesLock() *xtablesLock {
+	return &xtablesLock{lockFilePath: xtablesLockFilePath}
+}
+
+func (l *xtablesLock) acquire(timeout time.Duration) error {
+	var err error
+	var success bool
+
```

Comment:
do you want to check if `l.lockFile != nil` here, since you have the symmetric check in `release`?

Reply:
Sure, I found the current interface introduces some unnecessary error handling and race condition while the filelock can be thread-safe and process-thread. I changed the code to make it simple and thread-safe, PTAL.